File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/dtos/ContactInfoData.java
package dev.jefy.connectpro.portfolio.applicaion.dtos;


import dev.jefy.connectpro.portfolio.domain.vo.ContactInfo;
import dev.jefy.connectpro.user.domain.vo.Email;

/**
 * @author Jôph Yamba
 */
public record ContactInfoData(
        String email,
        String phone1,
        String phone2,
        String websiteUrl
) {
    public static ContactInfoData from(ContactInfo contactInfo) {
        return new ContactInfoData(
                contactInfo.email().value(),
                contactInfo.phone1(),
                contactInfo.phone2(),
                contactInfo.websiteUrl()
        );
    }

    public ContactInfo toDomain(){
        return new ContactInfo(
                new Email(this.email),
                this.phone1,
                this.phone2,
                this.websiteUrl
        );
        
    }
}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/dtos/ContactInfoData.kt
package dev.jefy.connectpro.portfolio.application.dtos

import dev.jefy.connectpro.portfolio.domain.vo.ContactInfo
import dev.jefy.connectpro.user.domain.vo.Email


/**
 * @author Jôph Yamba
 */
data class ContactInfoData(val email: String, val phone1: String, val phone2: String?, val websiteUrl: String?) {
    fun toContactInfo(): ContactInfo = ContactInfo(
        email = Email.of(this.email),
        phone1 = this.phone1,
        phone2 = this.phone2,
        websiteUrl = this.websiteUrl
    )
}

fun ContactInfo.toData(): ContactInfoData = ContactInfoData(
    email = this.email.value,
    phone1 = this.phone1,
    phone2 = this.phone2,
    websiteUrl = this.websiteUrl
)